src/functions.php

```diff
@@ -44,3 +45,42 @@ function listClassesInDirectory(string $directory): iterable
 {
     yield from listClassesInDirectories($directory);
 }
+
+/**
+ * get a list of all classes in the given file.
+ *
+ * @param string $file
+ *
+ * @return iterable
+ */
+function listClassesInFile(string $file): iterable
+{
+    $sourceLocator = new AggregateSourceLocator([
+        new SingleFileSourceLocator(
+            $file,
+            (new BetterReflection())->astLocator()
+        ),
+        // ↓ required to autoload parent classes/interface from another directory (e.g. /vendor)
+        new AutoloadSourceLocator((new BetterReflection())->astLocator()),
+    ]);
+
+    foreach ((new ClassReflector($sourceLocator))->getAllClasses() as $class) {
+        yield $class->getName();
+    }
+}
+
+/**
+ * get a list of all classes in the given files.
+ *
+ * @param string[] $files
+ *
+ * @return iterable
+ */
+function listClassesInFiles(string ...$files): iterable
+{
+    foreach ($files as $file) {
+        foreach (listClassesInFile($file) as $class) {
```

Comment:
You could use `yield from` here instead: 

```suggestion
        yield from listClassesInFile($file);
```

Reply:
:+1: